pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java

```diff
@@ -398,6 +390,58 @@ public boolean isFixedWidth() {
     public boolean isNumeric() {
       return this == INT || this == LONG || this == FLOAT || this == DOUBLE;
     }
+
+    /**
+     * Converts the given string value to the data type. Returns byte[] for BYTES.
+     */
+    public Object convert(String value) {
+      try {
+        switch (this) {
+          case INT:
+            return Integer.valueOf(value);
+          case LONG:
+            return Long.valueOf(value);
+          case FLOAT:
+            return Float.valueOf(value);
+          case DOUBLE:
+            return Double.valueOf(value);
+          case STRING:
+            return value;
+          case BYTES:
+            return BytesUtils.toBytes(value);
+          default:
+            throw new IllegalStateException();
```

Comment:
can also return a single element array if `isSingleValueField` is false for `this` ?

Reply:
This is on `DataType` level, where `isSingleValueField` is not accessible. That logic is handled within the `FieldSpec`